docs/source/tutorials/06_using_your_own_models.rst

```diff
@@ -0,0 +1,578 @@
+               """
+               img = cv2.cvtColor(inputs["img"], cv2.COLOR_BGR2RGB)
+               img = cv2.resize(img, (IMG_WIDTH, IMG_HEIGHT))
+               img = np.expand_dims(img, axis=0)
+               predictions = self.model.predict(img)
+               score = tf.nn.softmax(predictions[0])
+
+               return {
+                     "pred_label": self.class_label_map[np.argmax(score)],
+                     "pred_score": 100.0 * np.max(score),
+               }
+
+The custom node takes in the in-built PeekingDuck :term:`img` data type, makes a prediction based
+on the image, and produces two custom data types: `pred_label`, the predicted label ("defective"
+or "normal"); and `pred_score`, which is the confidence score of the prediction.
+
+
+Using the Classifier in a PeekingDuck Pipeline
+----------------------------------------------
+
+We'll now pair this custom node with other PeekingDuck nodes to build a complete solution. Imagine
+an automated inspection system like the one shown below, where the castings are placed on a
+conveyor belt and a camera takes a picture of each casting and sends it to the PeekingDuck pipeline
+for prediction. A report showing the predicted result for each casting is produced for the quality
+inspector, and he/she can use the report for further analysis. 
+
+   .. figure:: /assets/tutorials/automated_inspection.png
+      :width: 416
+      :alt: Vision Based Inspection of Conveyed Objects
+
+      Vision Based Inspection of Conveyed Objects (Source: `ScienceDirect <https://www.sciencedirect.com/science/article/pii/S221282711200248X>`_)
+
+Edit the ``pipeline_config.yml`` file to use the :mod:`input.visual` node to read in the images,
+and the :mod:`output.csv_writer` node to produce the report. We will test our solution on the 10 
+casting images in ``castings_data/inspection``, where each image's filename is a unique casting ID
+such as ``28_4137.jpeg``.
+
+**pipeline_config.yml**:
+
+   ``pipeline_config.yml`` updated content:
+
+   .. code-block:: yaml
+      :linenos:
+
+      nodes:
+      - input.visual:
+         source: castings_data/inspection
+      - custom_nodes.model.casting_classifier
+      - output.csv_writer:
+         stats_to_track: ["filename", "pred_label", "pred_score"]
+         file_path: casting_predictions.csv
+         logging_interval: 0
+
+   | Line 2 :mod:`input.visual`: tells PeekingDuck to load the images from 
+            ``castings_data/inspection``.
+   | Line 4 Calls the custom model node that you have just created.
+   | Line 5 :mod:`output.csv_writer`: produces the report for the quality inspector in a CSV file
+            (``castings_predictions.csv``). This node receives the :term:`filename` data type from
+            :mod:`input.visual`, the custom data types `pred_label` and `pred_score` from the 
+            custom model node, and writes them to the columns of the CSV file.
+
+Run the above with the command :greenbox:`peekingduck run`. |br|
```

Comment:
I got the following PeekingDuck runtime warning on macOS:
```
2022-05-04 13:25:04 peekingduck.pipeline.nodes.output.csv_writer  WARNING:  [] are not valid outputs.
Data pool only has this outputs: ['img', 'filename', 'pipeline_end', 'saved_video_fps', 'pred_label', 'pred_score']
Only ['filename', 'pred_label', 'pred_score'] will be logged in the csv file
```
Not sure if this is supposed to be the "normal" behavior.

Also, the output csv file is actually named `casting_predictions_DDMMYY-hh-mm-ss.csv` instead of `casting_predictions.csv` as mentioned above.

Reply:
There was a logic error in `output.csv_writer` that caused this to be logged when it shouldn't, have fixed it.
Added a small note that a time stamp would be appended.